extensions/typescript-language-features/src/typescriptServiceClient.ts

```diff
@@ -952,6 +957,47 @@ export default class TypeScriptServiceClient extends Disposable implements IType
 			this.executeWithoutWaitingForResponse('configurePlugin', { pluginName, configuration });
 		}
 	}
+
+	private notifyExtendedLogging(context: vscode.ExtensionContext) {
+		const getLastChange = (): Date | undefined => {
+			const lastChange = context.globalState.get<string | undefined>('typescript.tsserver.logLevelChanged');
+
+			if (lastChange) {
+				const date = new Date(lastChange);
+				if (date instanceof Date && !isNaN(date.valueOf())) {
+					return date;
+				}
+			}
+			return undefined;
+		};
+
+		const logLevel = this._configuration.tsServerLogLevel;
+		const lastLogLevelChange = getLastChange();
+
+		if (logLevel !== TsServerLogLevel.Off && lastLogLevelChange) {
+			const lastChangePlusOneWeek = new Date(lastLogLevelChange.valueOf() + /* 7 days in milliseconds */ 86400000 * 7);
+
+			if (lastChangePlusOneWeek < new Date()) {
+				vscode.window.showInformationMessage<vscode.MessageItem>(
+					localize(
+						'typescript.openTsServerLog.loggingEnabledForExtendedPeriod',
+						'TS Server logging has been enabled for an extended period. Please set `typescript.tsserver.log` to `off` and restart the TS server to disable logging'),
```

Comment:
For the message, let's explain why having logging enabled matters, for example:

`TS Server logging is currently enabled which may impact performance` 

which could show with the buttons:

- `Disable logging`
- `Don't show again`